### PR TITLE
engine: return InternalRows instead of Rows

### LINKED ISSUE #182: Return already converted Rows (a.k.a. RDD[InternalRow]) instead of standard Rows to improve speed performance.
On `BaseRelation` class we have:

```
  /**
   * Whether does it need to convert the objects in Row to internal representation, for example:
   *  java.lang.String to UTF8String
   *  java.lang.Decimal to Decimal
   *
   * If `needConversion` is `false`, buildScan() should return an `RDD` of `InternalRow`
   *
   * @note The internal representation is not stable across releases and thus data sources outside
   * of Spark SQL should leave this as true.
   *
   * @since 1.4.0
   */
  def needConversion: Boolean = true
```

we should override it to false, and change the output of CatalystScan from `RDD[Row]` to `RDD[InternalRow]`, changing all the generated row types to the converted types (per example, Date rows are Integers in InternalRows).

### PR DESCRIPTION
**Pull request by erizocosmico**
Closes #182

### CODE REVIEW

File: src/main/scala/tech/sourced/engine/iterator/CommitIterator.scala [hunk 1]
```diff
@@ -36,31 +36,42 @@ class CommitIterator(finalColumns: Array[String],
       filters.flatMap(_.matchingCases)
     )
 
-  /** @inheritdoc */
```
Comment (jfontan) [inline src/main/scala/tech/sourced/engine/iterator/CommitIterator.scala]: This and the opposite change is happening in the code. I'm not sure which is the standard way but I prefer spaces between comment marks.

Comment (erizocosmico) [reply src/main/scala/tech/sourced/engine/iterator/CommitIterator.scala]: Not intentionally made. It's just that damn IntelliJ removes the space when you autoformat, but when you autoformat again it adds it. Perhaps we should stick to multiline comments only so this does not happen.

File: src/main/scala/tech/sourced/engine/DefaultSource.scala [hunk 2]
```diff
@@ -130,7 +132,7 @@ case class GitRelation(session: SparkSession,
 
       // FIXME: when the RDD is persisted to disk the last element of this iterator is closed twice
       new CleanupIterator(iter.getOrElse(Seq().toIterator), provider.close(pds, repo))
-    })
+    }).asInstanceOf[RDD[Row]]
```
Comment (mcarmonaa) [inline src/main/scala/tech/sourced/engine/DefaultSource.scala]: From the `needConversion` documentation: `If needConversion is false, buildScan() should return an RDD of InternalRow`

Why are we returning a `RDD[Row]` instead or why shouldn't we return a `RDD[InternalRow]`?

Comment (mcarmonaa) [reply src/main/scala/tech/sourced/engine/DefaultSource.scala]: :+1:

**Comment by ajnavarro** [comment]
Did we get any improvement doing this? if not, is not worth it merge it.

**Comment by erizocosmico** [comment]
It's actually... slower 

wat

**Pull request closed without merging**